src/main/java/analysis/GoblintAnalysis.java

```diff
@@ -0,0 +1,189 @@
+        return "GobPie";
+    }
+
+
+    /**
+     * The method that is triggered to start a new analysis.
+     *
+     * @param files    the files that have been opened in the editor (not using due to using the compilation database).
+     * @param consumer the server which consumes the analysis results.
+     * @param rerun    tells if the analysis should be reran.
+     */
+
+    @Override
+    public void analyze(Collection<? extends Module> files, AnalysisConsumer consumer, boolean rerun) {
+        if (rerun) {
+            if (consumer instanceof MagpieServer) {
+
+                preAnalyse();
+
+                MagpieServer server = (MagpieServer) consumer;
+                if (reanalyse()) server.consume(new ArrayList<>(readResultsFromJson()), source());
+
+            }
+        }
+    }
+
+
+    /**
+     * The method that is triggered before each analysis.
+     * 
+     * preAnalyzeCommand is read from the GobPie configuration file.
+     * Can be used for automating the compilation database generation.
+     */
+
+    private void preAnalyse() {
+        String[] preAnalyzeCommand = goblintServer.getPreAnalyzeCommand();
+        if (preAnalyzeCommand != null) {
+            try {
+                runCommand(new File(System.getProperty("user.dir")), preAnalyzeCommand);
+            } catch (IOException | InvalidExitValueException | InterruptedException | TimeoutException e) {
+                this.magpieServer.forwardMessageToClient(
+                        new MessageParams(MessageType.Warning, "Running preanalysis command failed. " + e.getMessage()));
+            }
+        }
+    }
+
+
+    /**
+     * Sends the request to Goblint server to reanalyse.
+     *
+     * @return returns true if the request was sucessful, false otherwise
+     */
+
+    private boolean reanalyse() {
+
+        // {"jsonrpc":"2.0","id":0,"method":"analyze","params":{}}
+        String request = new GsonBuilder().create().toJson(new Request("analyze"));
+
+        try {
+            goblintClient.writeMessageToSocket(request);
+            return true;
+        } catch (IOException e) {
+            log.info("Sending the request to the server failed.");
+            return false;
+        }
+
+    }
+
+
+    /**
+     * Method for running a command.
+     *
+     * @param dirPath The directory in which the command will run.
+     * @param command The command to run.
+     * @return Exit value and output of a finished process.
+     */
+
+    public ProcessResult runCommand(File dirPath, String[] command) throws IOException, InvalidExitValueException, InterruptedException, TimeoutException {
+        log.debug("Waiting for command: " + command.toString() + " to run...");
+        ProcessResult process = new ProcessExecutor()
+                .directory(dirPath)
+                .command(command)
+                .redirectOutput(System.err)
+                .redirectError(System.err)
+                .execute();
+        return process;
+    }
```

Comment:
Unused now?

Reply:
No, it is used for running the pre-analyze command.